core/src/com/google/inject/internal/InjectionRequestProcessor.java

```diff
@@ -138,19 +138,34 @@ void validate() {
     }
 
     void injectMembers() {
-
-      // TODO(b/366058184): investigate a methodhandle version of this.
       InternalContext context = injector.enterContext();
       try {
         boolean isStageTool = injector.options.stage == Stage.TOOL;
         for (SingleMemberInjector memberInjector : memberInjectors) {
           // Run injections if we're not in tool stage (ie, PRODUCTION or DEV),
           // or if we are in tool stage and the injection point is toolable.
           if (!isStageTool || memberInjector.getInjectionPoint().isToolable()) {
-            try {
-              memberInjector.inject(context, null);
-            } catch (InternalProvisionException e) {
-              errors.merge(e);
+            if (InternalFlags.getUseExperimentalMethodHandlesOption()) {
+              try {
+                // In theory, constructing the handle to invoke it exactly once is expensive and wasteful, and it is true for
```

Comment:
Question: Why does this need to be a one-off instead of being reusable like the other paths?

Reply:
AFAICT this path is only hit once since this is for static injection, so there wouldn't be much of a benefit of combining all the handles together.